src/factory.js

```diff
@@ -61,11 +61,10 @@ export default function plotComponentFactory(Plotly) {
       this.getRef = this.getRef.bind(this);
       this.handleUpdate = this.handleUpdate.bind(this);
       this.figureCallback = this.figureCallback.bind(this);
+      this.updatePlotly = this.updatePlotly.bind(this);
     }
 
-    componentDidMount() {
-      this.unmounting = false;
-
+    updatePlotly(shouldInvokeResizeHandler, figureCallbackFunction, shouldAttachUpdateEvents) {
```

Comment:
for whatever reason, aesthetically it seems nicer to me to have the `should*` both at the end, but I can live with it as is of course.

Reply:
You mean the lifecycle methods? Yeah, I was on the fence here. I decided to do `updatePlotly` first, because it's called by those two...